map.js

```diff
@@ -11,70 +17,86 @@ let mapY = 0;
     wrapper.onmousedown = function(){mousedown = true;};
     wrapper.onmouseup = function(){mousedown = false;};
     wrapper.onmouseout = function(){mousedown = false;};
-    wrapper.onmousemove = function(e){
-        if(mousedown){
-            moveMap(e);
-        }
-    };
+    wrapper.onmousemove = function(e){if(mousedown){moveMap(e);}};
     wrapper.onwheel = function(e){zoomMap(e)};
 
     //attaches width to width of iframe
     window.addEventListener("mousemove", iFrameCheck);
 }
- 
-//attaches width to width of iframe
-function iFrameCheck(){
-    if(document.getElementById("iframeContainer")){
-        document.getElementById("iframeContainer").onclick = resizeMap;
-        window.removeEventListener("mousemove", iFrameCheck); //trim listener, it's served it's purpose.
-        resizeMap();
-    }
-}
-
-//matches width of map to Iframe width
-function resizeMap(){
-    if(document.getElementById("iframeContainer")){
-        var ifc = document.getElementById("iframeContainer");
-
-        if(debug){
-            ifc.style.width = "1000px";
-            ifc.style.height = "25px";
-            wrapper.style.height = "580px";
-        }
-
-        wrapper.style.width = ifc.clientWidth + "px";
-        wrapper.style.top = (ifc.clientHeight + 48).toString() + "px";
-    }
-    drawMap();    
-}
 
 function drawMap(){
     //maybe useful for placing icons.
     var aspectw = 443;
     var aspecth = 362;
 
-    mapCanvasContext.drawImage(img, mapX, mapY, (img.width * zoomLevel), (img.height * zoomLevel), 
-                                    0, 0, img.width, img.height);
+    mapCanvasContext.drawImage(img_Map, mapX, mapY, (img_Map.width * zoomLevel), (img_Map.height * zoomLevel), 
+                                    0, 0, img_Map.width, img_Map.height);
+    //draw all map icons //TODO: add in overlay options
+    for(let i = 0; i < locArr.length;i++){
+        if(locArr[i].icon == "Ayleid") drawIcon(icon_Ayleid, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Camp") drawIcon(icon_Camp, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Cave") drawIcon(icon_Cave, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Fort") drawIcon(icon_Fort, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Gate") drawIcon(icon_Gate, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Inn") drawIcon(icon_Inn, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Landmark") drawIcon(icon_Landmark, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Mine") drawIcon(icon_Mine, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Settlement") drawIcon(icon_Settlement, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Shrine") drawIcon(icon_Shrine, locArr[i].approxX, locArr[i].approxY);
+        else console.warn("Element at " + i +" has no icon.");
```

Comment:
error message should probably have more info about the element. What i've started doing is printing 2 lines: first the statement, and then the object. taht way the object shows up in the debugger and you can inspect it manually

Reply:
I don't think this error will ever get triggered since we already have all the icons and locations labeled with what icon goes to them. 

I left it in as a catchall to avoid null issues, and also made it to draw an X by default. Just in case something happens like a location.JSON rewrite or something.